c_plus_plus/src/model/recorder/recorder.cpp

```diff
@@ -0,0 +1,86 @@
+#include "recorder.h"
+
+#include <QCamera>
+#include <QCameraInfo>
+#include <QCameraViewfinder>
+#include <QMediaRecorder>
+#include <QUrl>
+
+namespace Model
+{
+
+Recorder::Recorder(const QString cameraDevice, QWidget *parent)
+    : IRecorder(parent)
+    , m_camera(new QCamera(getCameraInfo(cameraDevice), this))
+    , m_mediaRecorder(new QMediaRecorder(m_camera, this))
+{
+    m_camera->setCaptureMode(QCamera::CaptureMode::CaptureVideo);
+
+    QVideoEncoderSettings videoSettings;
+    videoSettings.setQuality(QMultimedia::VeryHighQuality);
+    videoSettings.setCodec("video/x-vp8");
+    m_mediaRecorder->setVideoSettings(videoSettings);
+
+    QAudioEncoderSettings audioSettings;
+    audioSettings.setQuality(QMultimedia::VeryHighQuality);
+    audioSettings.setCodec("audio/x-vorbis");
+    m_mediaRecorder->setAudioSettings(audioSettings);
+
+    m_mediaRecorder->setContainerFormat("video/webm");
+}
+
+void Recorder::start(const QString outputPath)
+{
+    m_mediaRecorder->setOutputLocation(QUrl::fromLocalFile(outputPath + "/media"));
```

Comment:
We should check the boolean output of setOutputLocation to be sure the setter worked.